### ISSUE TITLE
Embedded view disappear on layout reload

### ISSUE THREAD

**Issue by sebleclerc** [schibsted/layout#95]
Hi! I am reloading my layout using CMD+R, my embedded views disappear.

Here is my layout for my ViewController:
```
<UIView
    width="100%"
    height="100%">

    <UIView
        outlet="view1"
        height="60"
        backgroundColor="red" />

    <UIView
        outlet="view2"
        top="#view1.bottom"
        backgroundColor="green" />
</UIView>
```

Inside my ViewController, I have a reference to my custom collection view and I add his view to my ```view1```:
```
@IBOutlet weak var view1: UIView!
private let collectionView: SomeCustomCollectionView

view1.addSubview(collectionView.view)
```

Everything works as expected. I see my collection view loaded. 

When I hit CMD+R, the collection view disappear and I see the read background from my ```view1```. I check the view hierarchy and I see my collection view embedded inside a UIView and then the new layout on top of that.

I would think when the layout unload, the ```view1``` won't get deallocated because my collection view is not remove from the view hierarchy and then the whole layout gets created and added to the screen on top of what was left of the old layout.

Is there a way to be notified, inside a LayoutViewController, of the unloading of the layout? This way I could remove my collection view from the view hierarchy. Or what is the correct way to do this?

Thanks!

**Comment by nicklockwood** [comment]
Any programmatically-created views should be added inside `layoutDidLoad(_:)` instead of `viewDidLoad()`, that way they’ll be re-created after reload.

**Comment by sebleclerc** [comment]
Yep that is exactly what I am doing.

```
override func layoutDidLoad(_: LayoutNode) {
        embed(controller: collectionView, in: view1)
    }
```

```
func embed(controller: UIViewController, in container: UIView)
    {
        if !childViewControllers.contains(controller) {
            addChildViewController(controller)
        }

        controller.view.frame = container.frame

        if !view.subviews.contains(controller.view) {
            view.addSubview(controller.view)
        }

        controller.didMove(toParentViewController: self)
    }
```

**Comment by sebleclerc** [comment]
Hum, I think I have found what is wrong even tho I don't know why. For some I had added the 
``` if !childViewControllers.contains(controller) { ```
and
``` if !view.subviews.contains(controller.view) { ```

because I didn't wanted the same view to be added multiple times even if I call that util function multiple times. Now if I remove those verifications and always add child view controller and add subview... it works and the "old" view is removed from the hierarchy. I recall a view can't be part of 2 superview so probably when I ```addSubview``` it removes from the old UIView and added to the new and then the old UIView gets deallocated.
